data/transactions/verify/txn.go

```diff
@@ -218,10 +235,10 @@ func stxnVerifyCore(s *transactions.SignedTxn, txnIdx int, groupCtx *GroupContex
 			return nil
 		}
 
-		return errors.New("signedtxn has no sig")
+		return signatureErrorf("signedtxn has no sig")
 	}
 	if numSigs > 1 {
-		return errors.New("signedtxn should only have one of Sig or Msig or LogicSig")
+		return signatureErrorf("signedtxn should only have one of Sig or Msig or LogicSig")
```

Comment:
I would not annotate this as a SignatureError. It is, in some sense, a WellFormedness error. A caller who causes this is doing something really wrong - it's not just that they don't the keys for the txns they want to simulate.